bodhi/server/consumers/automatic_updates.py

```diff
@@ -0,0 +1,212 @@
+# Copyright © 2019 Red Hat, Inc. and others.
+#
+# This file is part of Bodhi.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+The Bodhi handler that creates updates automatically from tagged builds.
+
+This module is responsible for the process of creating updates when builds are
+tagged with certain tags.
+"""
+
+import logging
+import re
+import typing
+
+import fedora_messaging
+
+from bodhi.server import buildsys, initialize_db
+from bodhi.server.config import config
+from bodhi.server.exceptions import BodhiException
+from bodhi.server.models import Build, ContentType, Package, Release
+from bodhi.server.models import Update, UpdateType, User
+from bodhi.server.util import transactional_session_maker
+
+
+log = logging.getLogger('bodhi')
+
+
+class AutomaticUpdateHandler:
+    """
+    The Bodhi Automatic Update Handler.
+
+    A consumer that listens for messages about tagged builds and creates
+    updates from them.
+    """
+
+    _auto_update_cfg_key_re = re.compile(
+        r'(?P<releasename>\w+)\.autoupdates\.(?P<subkey>(?:\w+\.)*\w+)', re.ASCII)
+
+    _version_number_re = re.compile(r'[^\d]*(?P<vernum>\d+).*', re.ASCII)
+
+    def __init__(self, db_factory: transactional_session_maker = None):
+        """
+        Initialize the Automatic Update Handler.
+
+        Args:
+            db_factory (bodhi.server.util.TransactionalSessionMaker or None): If given, used as the
```

Comment:
We don't need to document the type here since you used the fancy new type hinting above. I'm actually in the middle of a (long) project to convert all of Bodhi to use type hinting instead of docblocks to document types.

```suggestion
            db_factory: If given, used as the
```

Reply:
I'll fix this (and reindent).